libs/azure-storage/langchain_azure_storage/document_loaders.py

```diff
@@ -0,0 +1,112 @@
+"""LangChain integrations for Azure Storage."""
+
+from typing import Callable, Iterable, Iterator, Optional, Union
+
+import azure.core.credentials
+import azure.core.credentials_async
+import azure.identity
+from azure.storage.blob import BlobClient, ContainerClient
+from langchain_core.document_loaders import BaseLoader
+from langchain_core.documents.base import Blob, Document
+
+SDK_CREDENTIAL_TYPE = Optional[
+    Union[
+        azure.identity.DefaultAzureCredential,
+        azure.core.credentials.AzureSasCredential,
+        azure.core.credentials.TokenCredential,
+    ]
+]
+
+
+def _get_client_kwargs(credential: SDK_CREDENTIAL_TYPE = None) -> dict:
+    return {"credential": credential, "connection_data_block_size": 256 * 1024}
+
+
+def _lazy_load_documents_from_blob(blob_client: BlobClient) -> Iterator[Document]:
+    blob_data = blob_client.download_blob(max_concurrency=10)
+    blob = Blob.from_data(
+        data=blob_data.readall(),
+        mime_type=blob_data.properties.content_settings.content_type,
+        metadata={
+            "source": blob_client.url,
+        },
+    )
+    yield Document(page_content=blob.as_string(), metadata=blob.metadata)
+
+
+def _get_sdk_credential(credential: SDK_CREDENTIAL_TYPE) -> SDK_CREDENTIAL_TYPE:
+    if credential is None:
+        return azure.identity.DefaultAzureCredential()
+    if isinstance(
+        credential,
+        (
+            azure.core.credentials.TokenCredential,
+            azure.core.credentials.AzureSasCredential,
+        ),
+    ):
+        return credential
+    raise TypeError("Invalid credential type provided.")
+
+
+class AzureBlobStorageLoader(BaseLoader):
+    """Document loader for loading LangChain Document objects from a container."""
+
+    def __init__(
+        self,
+        account_url: str,
+        container_name: str,
+        blob_names: Optional[Union[str, Iterable[str]]] = None,
+        prefix: str = "",
```

Comment:
Let's make `prefix` an `Optional[str]`. It's sort of nuanced but it will allow us to validate whether the customer provided a prefix and blob name as we can check whether the prefix is not `None` as opposed to falsey.